File: studyq-backend/routes/analytics_routes.py
from flask import Blueprint
from flask_jwt_extended import jwt_required, get_jwt_identity

from services.analytics_service import get_summary, get_history, get_trends
from utils.helpers import success_response

analytics_bp = Blueprint("analytics", __name__)


@analytics_bp.route("/summary", methods=["GET"])
@jwt_required()
def summary():
    user_id = int(get_jwt_identity())
    return success_response(get_summary(user_id))


@analytics_bp.route("/history", methods=["GET"])
@jwt_required()
def history():
    user_id = int(get_jwt_identity())
    return success_response(get_history(user_id))


@analytics_bp.route("/trends", methods=["GET"])
@jwt_required()
def trends():
    user_id = int(get_jwt_identity())
    return success_response(get_trends(user_id))

File: studyq-backend/routes/analyticsRoutes.js
const express = require('express');
const { getSummary, getHistory, getTrends } = require('../services/analyticsService');
const { successResponse, errorResponse } = require('../utils/helpers');
const authMiddleware = require('../middleware/auth');

const router = express.Router();

router.get('/summary', authMiddleware, async (req, res) => {
  try {
    const summary = await getSummary(req.user.id);
    return successResponse(res, summary);
  } catch (err) {
    console.error(err);
    return errorResponse(res, "Server error fetching summary.", 500);
  }
});

router.get('/history', authMiddleware, async (req, res) => {
  try {
    const history = await getHistory(req.user.id);
    return successResponse(res, history);
  } catch (err) {
    console.error(err);
    return errorResponse(res, "Server error fetching history.", 500);
  }
});

router.get('/trends', authMiddleware, async (req, res) => {
  try {
    const trends = await getTrends(req.user.id);
    return successResponse(res, trends);
  } catch (err) {
    console.error(err);
    return errorResponse(res, "Server error fetching trends.", 500);
  }
});

module.exports = router;
